src/ranged.cpp

```diff
@@ -745,11 +745,12 @@ dealt_projectile_attack player::throw_item( const tripoint &target, const item &
         proj_effects.insert( "SHATTER_SELF" );
     }
 
-    if( rng(0, 100) < 20 + skill_level * 12 && thrown.type->melee_cut > 0 ) {
-        const auto type =
-            ( thrown.has_flag("SPEAR") || thrown.has_flag("STAB") ) ?
-            DT_STAB : DT_CUT;
-        proj.impact.add_damage( type, thrown.type->melee_cut );
+    if( rng(0, 100) < 20 + skill_level * 12 ) {
+        int cut = thrown.damage_melee( DT_CUT );
+        int stab = thrown.damage_melee( DT_STAB );
+        if( cut || stab ) {
+            proj.impact.add_damage( cut > stab ? DT_CUT : DT_STAB, cut > stab ? cut : stab );
```

Comment:
Would be significantly more readable as
```
if( cut > 0 || stab > 0 ) {
    proj.impact.add_damage( cut > stab ? DT_CUT : DT_STAB, std::max( cut, stab ) );
}
```

Or even just by allowing stuff to deal both types separately.